src/plugins/intel_gpu/src/plugin/ops/generate_proposals.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "intel_gpu/primitives/generate_proposals.hpp"
+
+#include <ngraph_ops/generate_proposals_ie_internal.hpp>
+
+#include "intel_gpu/plugin/common_utils.hpp"
+#include "intel_gpu/plugin/program.hpp"
+#include "intel_gpu/primitives/mutable_data.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+static void CreateGenerateProposalsIEInternalOp(
+    Program& p,
+    const std::shared_ptr<ngraph::op::internal::GenerateProposalsIEInternal>& op) {
+    validate_inputs_count(op, {4});
+    if (op->get_output_size() != 3) {
+        IE_THROW() << "GenerateProposals requires 3 outputs";
+    }
+
+    auto inputs = p.GetInputPrimitiveIDs(op);
+    const auto& attrs = op->get_attrs();
+    const auto layer_type_name = layer_type_name_ID(op);
+    const auto layer_name = layer_type_name + ".out0";
+
+    // output 2 - scores
+    const auto mutable_precision_1 = op->get_output_element_type(1);
+    const auto output_shape_1 = op->get_output_shape(1);
+    const cldnn::layout mutable_layout_1{cldnn::element_type_to_data_type(mutable_precision_1),
+                                         cldnn::format::get_default_format(output_shape_1.size()),
+                                         tensor_from_dims(output_shape_1)};
+    cldnn::memory::ptr shared_memory_1{p.GetEngine().allocate_memory(mutable_layout_1)};
+
+    const auto mutable_id_w_1 = layer_type_name + "_md_write.1";
+    const cldnn::mutable_data mutable_prim_w_1{mutable_id_w_1, shared_memory_1};
+    p.add_primitive(*op, mutable_prim_w_1);
+    inputs.push_back(mutable_id_w_1);
+
+    // output 3 - roisNum
+    const auto output_shape_2 = op->get_output_shape(2);
+    const auto mutable_precision_2 = op->get_output_element_type(2);
+    const cldnn::layout mutable_layout_2{cldnn::element_type_to_data_type(mutable_precision_2),
+                                         cldnn::format::get_default_format(output_shape_2.size()),
+                                         tensor_from_dims(output_shape_2)};
+    cldnn::memory::ptr shared_memory_2{p.GetEngine().allocate_memory(mutable_layout_2)};
+
+    const auto mutable_id_w_2 = layer_type_name + "_md_write.2";
+    const cldnn::mutable_data mutable_prim_w_2{mutable_id_w_2, shared_memory_2};
+    p.add_primitive(*op, mutable_prim_w_2);
+    inputs.push_back(mutable_id_w_2);
+
+    const cldnn::generate_proposals prim{layer_name,
+                                         inputs[0],
```

Comment:
I think the inputs for this API is too redundant and can be bundled with a vector<primitive> inputs.

Reply:
Done.